lib/Controller/WopiController.php

```diff
@@ -174,6 +174,19 @@ public function checkFileInfo(string $fileId, string $access_token): JSONRespons
 			$response['UserPrivateInfo']['SignatureKey'] = $documentSigningKey;
 			$documentSigningCa = $this->config->getUserValue($wopi->getEditorUid(), 'richdocuments', 'documentSigningCa', '');
 			$response['UserPrivateInfo']['SignatureCa'] = $documentSigningCa;
+
+			$eSignatureBaseUrl = $this->config->getAppValue(Application::APPNAME, 'esignature_base_url');
+			if ($eSignatureBaseUrl !== '') {
+				$response['UserPrivateInfo']['ESignatureBaseUrl'] = $eSignatureBaseUrl;
+			}
+			$eSignatureClientId = $this->config->getAppValue(Application::APPNAME, 'esignature_client_id');
+			if ($eSignatureClientId !== '') {
+				$response['UserPrivateInfo']['ESignatureClientId'] = $eSignatureClientId;
```

Comment:
Confuses me a bit that this is in UserPrivateInfo, while it is not specific to the user and rather an instance wide setting.

Reply:
This is probably because of my limited experience with WOPI: do you suggest to move this to a toplevel field in the response or something else?

The eid secret is something that the cool server should have access to, but we intentionally don't push it to the client.

Thanks.

Reply:
Yes, top level or nested like this should be fine I think, but maybe worth to check if there are more strong opinions from others working on WOPI on your side.

Reply:
I would tend to go with a ServerPrivateInfo toplevel field to contain not just esign properties, but anything where it's important to configure the WOPI allowlist. As you suggest, I'll check with others on the Collabora side and get back to you on this.

Reply:
Sounds good 👍

Reply:
https://github.com/CollaboraOnline/online/pull/10734 now changes the COOL side so we'll expect this proposed ServerPrivateInfo top-level key from you for esign settings. This can host any other instance wide credentials in the future.